src/Symfony/Component/OptionsResolver/Debug/OptionResolverIntrospector.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver\Debug;
+
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @final
+ */
+class OptionResolverIntrospector
+{
+    private $optionsResolver;
+    private $get;
+    private $has;
+
+    public function __construct(OptionsResolver $optionsResolver)
+    {
+        $this->optionsResolver = $optionsResolver;
+
+        $this->get = \Closure::bind(function ($property, $option) {
+            return $this->{$property}[$option];
+        }, $optionsResolver, $optionsResolver);
+
+        $this->has = \Closure::bind(function ($property, $option) {
+            return array_key_exists($option, $this->{$property});
+        }, $optionsResolver, $optionsResolver);
+    }
+
+    /**
+     * @param string $option
+     *
+     * @return mixed|null
+     */
+    public function getDefault($option)
+    {
+        return $this->get('defaults', $option);
+    }
+
+    /**
+     * @param string $option
+     *
+     * @return bool True if a default value was set
+     */
+    public function hasDefault($option)
```

Comment:
This method is required to differentiate a default set to a `null` value from no default set at all.